torchtitan/experiments/kernels/moe/indices.py

```diff
@@ -170,19 +183,25 @@ def generate_permute_indices(
         To:   | E0 | E1 | E2 | E3 | E0 | E1 | E2 | E3 |
               |  4 |  2 |  1 |  3 |  1 |  2 |  3 |  4 |
     """
+
     # prefix sum to get start index of each expert (parallel scan kernel in future?)
     start_index_values = (
         torch.cumsum(tokens_per_expert_group, 0) - tokens_per_expert_group
     )
 
-    # chunk sizes for each expert
-    chunk_size_per_expert = tokens_per_expert_group.view(num_ranks, -1).sum(0)
+    # total tokens for each expert (sum over ranks)
+    total_tokens_per_expert = tokens_per_expert_group.view(num_ranks, -1).sum(0)
 
-    # align the chunk sizes (cdiv)
-    m_sizes = ((chunk_size_per_expert + alignment - 1) // alignment * alignment).to(
-        torch.int32
+    # pad out empty experts to alignment requirement
+    padded_total_tokens_per_expert = torch.where(
```

Comment:
clamp_min is the best for eager (it won't run additional `total_tokens_per_expert == 0` kernel and `fill` kernel for scalar argument), with torch.compile it doesn't matter.

Reply:
thanks @ngimel! Definitely interested in maximizing eager perf - have switched to clamp_min.